### PR TITLE
If lineages are in an inconsistent (non-deployed) state, deploy them

### PR DESCRIPTION
**Pull request by pde**
Fixes:  #3497. Probably also fixes: #3366.

### CODE REVIEW

**Comment by pde** [comment]
Still needs tests...

**Comment by pde** [comment]
Obviously if we support `deploy_before_expiry` in the future this will need to get slightly more complicated, but I think that would just be an extra test in this spot.

File: certbot/main.py [hunk 1]
```diff
@@ -155,34 +155,39 @@ def _handle_subset_cert_request(config, domains, cert):
...
+    :param storage.RenewableCert lineage:
 
     :returns: Tuple of (str action, cert_or_None) as per _treat_as_renewal
               action can be: "newcert" | "renew" | "reinstall"
     :rtype: tuple
 
     """
-    if renewal.should_renew(config, cert):
-        return "renew", cert
+    if lineage.has_pending_deployment():
```
Comment (schoen) [inline certbot/main.py]: I think this is the right way to do this!

Comment (schoen) [inline certbot/main.py]: I'm not sure that it's ideal that there's absolutely no way to override this behavior with a command-line flag, although I appreciate the argument that typically users would not want to do so. While this is a good default even in respect to our UI, maybe we should consider having a way to be more explicit about this (although that risks a cascade of `--no-really` flags).

File: certbot/main.py [hunk 2]
```diff
@@ -155,34 +155,39 @@ def _handle_subset_cert_request(config, domains, cert):
...
-    ).format(cert.configfile.filename, br=os.linesep)
+    ).format(lineage.configfile.filename, br=os.linesep)
 
     if config.verb == "run":
         keep_opt = "Attempt to reinstall this existing certificate"
     elif config.verb == "certonly":
         keep_opt = "Keep the existing certificate for now"
     choices = [keep_opt,
-               "Renew & replace the cert (limit ~5 per 7 days)"]
+               "Renew & replace the lineage (limit ~5 per 7 days)"]
```
Comment (schoen) [inline certbot/main.py]: @pde now says this should be "cert" for UI reasons.

**Comment by schoen** [comment]
LGTM but @pde has something to change for UI reasons. Should be fine to merge after that.

**Pull request merged**